pkg/controller/address_space_controller_deployer/deployer.go

```diff
@@ -0,0 +1,348 @@
+		_, err = deploymentClient.Update(&agentDeployment)
+		if err != nil {
+			return err
+		}
+		waitForReplicas(podClient, &agentDeployment, 0)
+		log.Info("Agent scaled down", "deployment", agentDeployment.Name)
+	}
+
+	// Get rid of API service so that it doesn't prevent upgrade
+	aggClient, err := aggregatorclient.NewForConfig(d.config)
+	if err != nil {
+		return err
+	}
+	apiServiceClient := aggClient.ApiregistrationV1().APIServices()
+
+	log.Info("Deleting v1beta1.enmasse.io api service")
+	err = deleteApiServiceIfPresent(apiServiceClient, "v1beta1.enmasse.io")
+	if err != nil {
+		return err
+	}
+
+	log.Info("Deleting v1alpha1.enmasse.io api service")
+	err = deleteApiServiceIfPresent(apiServiceClient, "v1alpha1.enmasse.io")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deleteApiServiceIfPresent(client apiserviceclient.APIServiceInterface, name string) error {
+	_, err := client.Get(name, metav1.GetOptions{})
+	if err != nil {
+		if k8errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	dopts := metav1.DeleteOptions{}
+	return client.Delete(name, &dopts)
+}
+
+func applyDeployment(deployment *appsv1.Deployment) error {
+	install.ApplyDeploymentDefaults(deployment, "address-space-controller", deployment.Name)
+	install.ApplyContainer(deployment, "address-space-controller", func(container *corev1.Container) {
+		install.ApplyContainerImage(container, "address-space-controller", nil)
+		install.ApplyHttpProbe(container.LivenessProbe, 30, "/healthz", 8080)
+		install.ApplyHttpProbe(container.ReadinessProbe, 30, "/healthz", 8080)
+		container.Ports = []corev1.ContainerPort{
+			{Name: "http", ContainerPort: 8080, Protocol: corev1.ProtocolTCP},
+		}
+		install.ApplyEnvSimple(container, "JAVA_OPTS", "-verbose:gc")
+		install.ApplyEnvSimple(container, "ENABLE_EVENT_LOGGER", "true")
+		install.ApplyEnvSimple(container, "TEMPLATE_DIR", "/opt/templates")
+		install.ApplyEnvSimple(container, "RESOURCES_DIR", "/opt")
+		t := true
+		install.ApplyEnvConfigMap(container, "WILDCARD_ENDPOINT_CERT_SECRET", "wildcardEndpointCertSecret", "address-space-controller-config", &t)
+		install.ApplyEnvConfigMap(container, "RESYNC_INTERVAL", "resyncInterval", "address-space-controller-config", &t)
+		install.ApplyEnvConfigMap(container, "RECHECK_INTERVAL", "recheckInterval", "address-space-controller-config", &t)
+
+		install.ApplyEnvSimple(container, "IMAGE_PULL_POLICY", string(images.PullPolicyFromImageName(container.Image)))
+		applyImageEnv(container, "ROUTER_IMAGE", "qdrouterd-base")
+		applyImageEnv(container, "STANDARD_CONTROLLER_IMAGE", "standard-controller")
+		applyImageEnv(container, "AGENT_IMAGE", "agent")
+		applyImageEnv(container, "BROKER_IMAGE", "broker")
+		applyImageEnv(container, "BROKER_PLUGIN_IMAGE", "broker-plugin")
+		applyImageEnv(container, "TOPIC_FORWARDER_IMAGE", "topic-forwarder")
+		applyImageEnv(container, "MQTT_GATEWAY_IMAGE", "mqtt-gateway")
+		applyImageEnv(container, "MQTT_LWT_IMAGE", "mqtt-lwt")
+
```

Comment:
We need to pass FS_GROUP_OVERRIDE environment to the address space controller.  There's a function to do this already - pkg/util/install/utils.go:512